src/v/cluster/partition_balancer_planner.cc

```diff
@@ -682,26 +707,28 @@ auto partition_balancer_planner::request_context::do_with_partition(
     return visitor(part);
 }
 
-void partition_balancer_planner::request_context::for_each_partition(
+ss::future<> partition_balancer_planner::request_context::for_each_partition(
   ss::noncopyable_function<ss::stop_iteration(partition&)> visitor) {
-    for (const auto& t : _parent._state.topics().topics_map()) {
-        for (const auto& a : t.second.get_assignments()) {
-            auto ntp = model::ntp(t.first.ns, t.first.tp, a.id);
+    const auto& topics = _parent._state.topics();
+    for (auto it = topics_begin(topics); it != topics_end(topics); ++it) {
+        for (const auto& a : it->second.get_assignments()) {
```

Comment:
we should maybe_yield in this loop body too, a case with a few topics, each with a lot of partitions, is possible too

Reply:
Its there below (was done in a previous commit).. this commit just repalced the iterator.

Reply:
Hmm I'm not sure I understand - the whole per-partition loop for a single topic is done in a single continuation, right?